### PR TITLE
Add definition of uniqueness.

### PR DESCRIPTION
**Pull request by Thom1729**
For https://github.com/yaml/yaml-spec/pull/114#pullrequestreview-741107701.

I kept it minimal here, but we can add more if we think it's needed.

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -1098,6 +1098,10 @@ scheme.
 that such comparison would yield the correct results.
 
 
+##### Uniqueness
+
+> A [mapping's] [keys] are _unique_ if no two keys are equal to each other.
```
Comment (ingydotnet) [inline spec/spec.md:1103]: Should this be `no two keys are equal or identical to each other.` ?

Comment (eemeli) [reply spec/spec.md:1103]: I'm pretty sure that identity implies equality sufficiently well for it not to need to be separately specified.

**Review by eemeli — APPROVED** [review]

**Review by ingydotnet — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

File: spec/spec.md [hunk 2]
```diff
@@ -1098,6 +1098,10 @@ scheme.
...
+
```
Comment (ingydotnet) [inline spec/spec.md:1104]: Heading lines are preceded by 2 blank lines throughout the spec.md,

Comment (ingydotnet) [reply spec/spec.md:1104]: Fixed it myself...

**Pull request merged**